### PR TITLE
Invitation+Suggestion count in OrganizationSwitcherTrigger

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

This PR implements the below UI
image

### CODE REVIEW

File: packages/clerk-js/src/ui/primitives/NotificationBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  defaultVariants: {
+    colorScheme: 'primary',
+    textVariant: 'extraSmallRegular',
+  },
+}));
+
+// @ts-ignore
+export type NotificationBadgeProps = PropsOfComponent<typeof Flex> & StyleVariants<typeof applyVariants>;
+
+export const NotificationBadge = (props: NotificationBadgeProps) => {
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/primitives/NotificationBadge.tsx:35]: Nitpicky: 
```suggestion
export const Badge = (props: BadgeProps) => {
```

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx [hunk 2]
```diff
@@ -1,29 +1,71 @@
...
     const { username, primaryEmailAddress, primaryPhoneNumber, ...userWithoutIdentifiers } = useCoreUser();
     const { organization } = useCoreOrganization();
     const { hidePersonal } = useOrganizationSwitcherContext();
 
+    /**
+     * Prefetch user invitations and suggestions
+     */
+    const { userInvitations, userSuggestions } = useCoreOrganizationList(organizationListParams);
+
+    const notificationCount = (userInvitations.count ?? 0) + (userSuggestions.count ?? 0);
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx:59]:
```suggestion
    const notificationCount = (userInvitations.count || 0) + (userSuggestions.count || 0);
```

as we don't really care about `false`, `undefined` or `null` values here

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx [hunk 3]
```diff
@@ -1,29 +1,71 @@
...
+
+    const notificationCountAnimated = useDelayedVisibility(notificationCount, 350);
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx:61]: Not part of this PR, but can we please add a comment explaining what `useDelayedVisibility` does in `useDelayedVisibility.ts`?

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx [hunk 4]
```diff
@@ -1,29 +1,71 @@
...
+import { organizationListParams } from './utils';
 
 type OrganizationSwitcherTriggerProps = PropsOfComponent<typeof Button> & {
   isOpen: boolean;
 };
 
+function useEnterAnimation() {
+  const prefersReducedMotion = usePrefersReducedMotion();
+
+  const getFormTextAnimation = useCallback(
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx:24]: This helper looks more complicated than I'd expect. 
Do we need the useCallback here? Is there any performance issue or side-effect if we don't wrap the handler with it? In general, let's prefer not using useCallback or useMemo unless needed. 

If we manage to remove useCallback, I'd probably inline this behavior inside a different `NotificationCountBadge` component that would include the animation logic as well

File: packages/clerk-js/src/ui/primitives/NotificationBadge.tsx (hunk 1 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/primitives/NotificationBadge.tsx:35]: We already have a component named Badge, wouldn't that be confusing ?

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx (hunk 4 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx]: I removed useCallback. I created NotificationCountBadge but it is still nested in OrgSwitcher. Designs still don't indicate that this will be used globally or that the animation will always be used

Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx]: Thats great :)

File: packages/clerk-js/src/ui/hooks/useDelayedVisibility.ts [hunk 5]
```diff
@@ -1,5 +1,10 @@
 import { useEffect, useState } from 'react';
 
+/**
+ * Utility hook for delaying mounting of components for enter and exit animations.
+ * Delays to update the state when is switched from/to undefined.
+ * Immediate change for in-between changes
+ */
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/hooks/useDelayedVisibility.ts:7]: 🙏🏻

**Review by nikosdouvlis — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
